remove no longer needed lemmas (as of mathcomp-1.11)

chdoc opened the pull request.

chdoc committed:
remove no longer needed lemmas (as of mathcomp-1.11)

chdoc commented:
@palmskog , this drops `mathcomp-1.9` and mathcomp-1.10`, but I think three versions of mathcomp are enough. If we make a release with theory changes, maybe it's a good idea to include this.

palmskog commented:
@chdoc dropping 1.9 and 1.10 is perfectly fine by me, now that we have a release that is likely to work for a while on 1.13. Should I modify `meta.yml` and the CI configuration in the PR branch?

chdoc commented:
> @chdoc dropping 1.9 and 1.10 is perfectly fine by me, now that we have a release that is likely to work for a while on 1.13. Should I modify `meta.yml` and the CI configuration in the PR branch?

Yes, please do.

palmskog committed:
drop MathComp 1.9 and 1.10 from opam and ci

chdoc committed:
remove take_take and Sub_eq

chdoc committed:
trim misc.v

- [all/ex]1s -> [forall/exists]_cons (as in mathcomp-1.13)
- remove global hints for mem_head and ltn_ord

chdoc commented:
As far as I am concerned, this is ready to be merged.

palmskog commented:
Hmm, I think we could also bump to 8.15 compatibility, and add at least Nix CI for 8.15 (the MC 8.15 Docker images are not available yet). Or should we do 8.15 in a separate PR?

chdoc commented:
I think that should be a separate PR. Also, I don't expect any changes to the .v files for 8.15, given that master checks just fine.
